### PR TITLE
[bugfix] kvsdb: do not increment the counter if key exists

### PR DESCRIPTION
**Pull request by kevindiu**
Signed-off-by: kevindiu <kevin_diu@yahoo.com.hk>

This PR fixes a bug if the same uuid is added to the kvsdb, the counter will be increase and it will not match with the data size.

### Environment:

- Go Version: 1.16.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [x] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/core/ngt/service/kvs/uo.go [hunk 1]
```diff
@@ -116,36 +116,101 @@ func (e *entryUo) storeLocked(i *uint32) {
...
+		if e.unexpungeLocked() {
+			m.dirty[key] = e
+		}
+		actual, loaded, _ = e.tryLoadOrStore(value)
+	} else if e, ok := m.dirty[key]; ok {
+		actual, loaded, _ = e.tryLoadOrStore(value)
+		m.missLocked()
+	} else {
+		if !read.amended {
+
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/uo.go:140]:
```suggestion
```

**Review by hlts2 — APPROVED** [review]
thank you.
LGTM

File: pkg/agent/core/ngt/service/kvs/uo.go [hunk 2]
```diff
@@ -116,36 +116,100 @@ func (e *entryUo) storeLocked(i *uint32) {
...
 	read, _ := m.read.Load().(readOnlyUo)
 	e, ok := read.m[key]
 	if !ok && read.amended {
 		m.mu.Lock()
 		read, _ = m.read.Load().(readOnlyUo)
 		e, ok = read.m[key]
 		if !ok && read.amended {
+			e, ok = m.dirty[key]
 			delete(m.dirty, key)
+
```
Comment (kpango) [inline pkg/agent/core/ngt/service/kvs/uo.go:185]:
```suggestion
```

**Comment by kevindiu** [comment]
LGTM!

**Review by kpango — DISMISSED** [review]
LGTM.
can I push format commit befor merge?

**Comment by kevindiu** [comment]
yes, please :)

**Comment by kevindiu** [comment]
LGTM

**Review by hlts2 — APPROVED** [review]
LGTM

**Pull request merged**